From 24c31dc89ee212d41237db5f92df6c46d3d41167 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 19 Dec 2014 19:41:10 +0100
Subject: [PATCH] replace cjson with json-c

* debug.h: Rename to powerd_debug.h
  json-c is also staging debug.h and wrong one is used in builds where
  json-c is used instead of cjson.

  json-c include paths appear before include/internal and build fails
  with:
    powerd/4.0.0-25-r3/git/libpowerd/wait.c:126: undefined reference to
      `TRACE'
    collect2: error: ld returned 1 exit status

Change-Id: I2681bc0c50ecb3dd4ef97070ae4df434a0cf0a76
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 CMakeLists.txt                  |   6 +-
 README.md                       |   2 +-
 include/internal/debug.h        | 119 ----------------------------------------
 include/internal/powerd_debug.h | 119 ++++++++++++++++++++++++++++++++++++++++
 libpowerd/CMakeLists.txt        |   2 +-
 libpowerd/clock.c               |   2 +-
 libpowerd/commands.c            |   2 +-
 libpowerd/init.c                |   2 +-
 libpowerd/wait.c                |   2 +-
 powerd/CMakeLists.txt           |   2 +-
 powerd/alarms/timeout_alarm.c   |   2 +-
 powerd/charging/battery.c       |   4 +-
 powerd/charging/charger.c       |   4 +-
 powerd/charging/suspend.c       |   4 +-
 powerd/main.c                   |   2 +-
 powerd/suspend/shutdown.c       |   4 +-
 powerd/suspend/suspend_ipc.c    |   4 +-
 powerd/utils/init.c             |   2 +-
 powerd/utils/sysfs.c            |   2 +-
 powerd/utils/uevent.c           |   2 +-
 20 files changed, 144 insertions(+), 144 deletions(-)
 delete mode 100644 include/internal/debug.h
 create mode 100644 include/internal/powerd_debug.h

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b403a78..3d49758 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,9 +39,9 @@ pkg_check_modules(LUNASERVICE2 REQUIRED luna-service2)
 include_directories(${LUNASERVICE2_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${LUNASERVICE2_CFLAGS_OTHER})
 
-pkg_check_modules(CJSON REQUIRED cjson)
-include_directories(${CJSON_INCLUDE_DIRS})
-webos_add_compiler_flags(ALL ${CJSON_CFLAGS_OTHER})
+pkg_check_modules(JSON REQUIRED json-c)
+include_directories(${JSON_INCLUDE_DIRS})
+webos_add_compiler_flags(ALL ${JSON_CFLAGS_OTHER})
 
 pkg_check_modules(NYXLIB REQUIRED nyx)
 include_directories(${NYXLIB_INCLUDE_DIRS})
diff --git a/README.md b/README.md
index ab70a53..64fd999 100644
--- a/README.md
+++ b/README.md
@@ -20,7 +20,7 @@ Below are the tools and libraries (and their minimum versions) required to build
 * gcc 4.6.3
 * glib-2.0 2.32.1
 * make (any version)
-* openwebos/cjson 1.8.0
+* json-c 0.11
 * openwebos/cmake-modules-webos 1.0.0 RC3
 * openwebos/luna-service2 3.0.0
 * openwebos/nyx-lib 2.0.0 RC2
diff --git a/include/internal/debug.h b/include/internal/debug.h
deleted file mode 100644
index 0f62605..0000000
--- a/include/internal/debug.h
+++ /dev/null
@@ -1,119 +0,0 @@
-/* @@@LICENSE
-*
-*      Copyright (c) 2007-2013 LG Electronics, Inc.
-*
-* Licensed under the Apache License, Version 2.0 (the "License");
-* you may not use this file except in compliance with the License.
-* You may obtain a copy of the License at
-*
-* http://www.apache.org/licenses/LICENSE-2.0
-*
-* Unless required by applicable law or agreed to in writing, software
-* distributed under the License is distributed on an "AS IS" BASIS,
-* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-* See the License for the specific language governing permissions and
-* limitations under the License.
-*
-* LICENSE@@@ */
-
-
-#ifndef _DEBUG_H_
-#define _DEBUG_H_
-
-
-#include <stdio.h>
-#include <assert.h>
-#include <glib.h>
-#include <string.h>
-#include <stdbool.h>
-
-/* define this to assert when BUG() macro is called */
-#define ASSERT_ON_BUG
-
-#undef G_LOG_DOMAIN
-#define G_LOG_DOMAIN "Powerd"
-
-// TODO use common error codes
-#define FATAL_ERROR -1
-#define ERROR       1
-#define POWERD_ERROR_TIMEOUT 355
-
-/* Set the current log level */
-void SetLogLevel(GLogLevelFlags newLogLevel);
-
-/* Set the destination of the log */
-void SetUseSyslog(int useSyslog);
-
-/* Return the current log level */
-GLogLevelFlags GetLogLevel();
-
-int DebugInit(void);
-
-void _good_assert(const char * cond_str, bool cond);
-
-#ifdef ASSERT_ON_BUG
-#define BUG() {                     \
-    *( (int*) NULL) = 0;            \
-}
-#else
-#define BUG() {}
-#endif
-
-#define _assert(cond)               \
-do {                                \
-    _good_assert(#cond, cond);      \
-} while (0)
-
-#define g_critical_once(...)        \
-do {                                \
-    static int seen = 0;            \
-    if (!seen)                      \
-    {                               \
-        seen = 1;                   \
-        g_critical(__VA_ARGS__);    \
-    }                               \
-} while(0)
-
-#define g_info(...)                 \
-    g_log (G_LOG_DOMAIN,            \
-           G_LOG_LEVEL_INFO,        \
-           __VA_ARGS__)
-
-#define g_perror(...)           \
-do {                                   \
-    char buf[512];                     \
-    buf[0] = 0;                        \
-    g_critical(__VA_ARGS__);           \
-    strerror_r(errno, buf, 512);       \
-    g_error(buf); \
-} while(0)
-
-#define MESSAGE(...)                   \
-do {                                \
-    g_message(__VA_ARGS__);         \
-} while (0)
-
-#define TRACE(...)                  \
-do {                                \
-    g_debug(__VA_ARGS__);         \
-} while(0)
-
-#define TRACE_ERROR(...)  \
-do {                      \
-    g_error(__VA_ARGS__);        \
-} while(0)
-
-#define TRACE_PERROR g_perror
-
-#define iferr(cond, text, goto_label) \
-do {                                    \
-    if (cond)                           \
-    {                                   \
-        TRACE_ERROR(text);              \
-        goto goto_label;                \
-    }                                   \
-} while (0);
-
-void print_trace(void);
-
-#endif //_DEBUG_H_
diff --git a/include/internal/powerd_debug.h b/include/internal/powerd_debug.h
new file mode 100644
index 0000000..0f62605
--- /dev/null
+++ b/include/internal/powerd_debug.h
@@ -0,0 +1,119 @@
+/* @@@LICENSE
+*
+*      Copyright (c) 2007-2013 LG Electronics, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* LICENSE@@@ */
+
+
+#ifndef _DEBUG_H_
+#define _DEBUG_H_
+
+
+#include <stdio.h>
+#include <assert.h>
+#include <glib.h>
+#include <string.h>
+#include <stdbool.h>
+
+/* define this to assert when BUG() macro is called */
+#define ASSERT_ON_BUG
+
+#undef G_LOG_DOMAIN
+#define G_LOG_DOMAIN "Powerd"
+
+// TODO use common error codes
+#define FATAL_ERROR -1
+#define ERROR       1
+#define POWERD_ERROR_TIMEOUT 355
+
+/* Set the current log level */
+void SetLogLevel(GLogLevelFlags newLogLevel);
+
+/* Set the destination of the log */
+void SetUseSyslog(int useSyslog);
+
+/* Return the current log level */
+GLogLevelFlags GetLogLevel();
+
+int DebugInit(void);
+
+void _good_assert(const char * cond_str, bool cond);
+
+#ifdef ASSERT_ON_BUG
+#define BUG() {                     \
+    *( (int*) NULL) = 0;            \
+}
+#else
+#define BUG() {}
+#endif
+
+#define _assert(cond)               \
+do {                                \
+    _good_assert(#cond, cond);      \
+} while (0)
+
+#define g_critical_once(...)        \
+do {                                \
+    static int seen = 0;            \
+    if (!seen)                      \
+    {                               \
+        seen = 1;                   \
+        g_critical(__VA_ARGS__);    \
+    }                               \
+} while(0)
+
+#define g_info(...)                 \
+    g_log (G_LOG_DOMAIN,            \
+           G_LOG_LEVEL_INFO,        \
+           __VA_ARGS__)
+
+#define g_perror(...)           \
+do {                                   \
+    char buf[512];                     \
+    buf[0] = 0;                        \
+    g_critical(__VA_ARGS__);           \
+    strerror_r(errno, buf, 512);       \
+    g_error(buf); \
+} while(0)
+
+#define MESSAGE(...)                   \
+do {                                \
+    g_message(__VA_ARGS__);         \
+} while (0)
+
+#define TRACE(...)                  \
+do {                                \
+    g_debug(__VA_ARGS__);         \
+} while(0)
+
+#define TRACE_ERROR(...)  \
+do {                      \
+    g_error(__VA_ARGS__);        \
+} while(0)
+
+#define TRACE_PERROR g_perror
+
+#define iferr(cond, text, goto_label) \
+do {                                    \
+    if (cond)                           \
+    {                                   \
+        TRACE_ERROR(text);              \
+        goto goto_label;                \
+    }                                   \
+} while (0);
+
+void print_trace(void);
+
+#endif //_DEBUG_H_
diff --git a/libpowerd/CMakeLists.txt b/libpowerd/CMakeLists.txt
index 88bfffc..d5b5ab5 100644
--- a/libpowerd/CMakeLists.txt
+++ b/libpowerd/CMakeLists.txt
@@ -27,5 +27,5 @@ webos_build_pkgconfig()
 webos_add_compiler_flags(ALL -fPIC -DSTACK_GROWS_DOWN)
 
 add_library(libpowerd SHARED clock.c commands.c init.c wait.c)
-target_link_libraries(libpowerd ${GLIB2_LDFLAGS} ${LUNASERVICE2_LDFLAGS} ${CJSON_LDFLAGS} pthread rt)
+target_link_libraries(libpowerd ${GLIB2_LDFLAGS} ${LUNASERVICE2_LDFLAGS} ${JSON_LDFLAGS} pthread rt)
 webos_build_library(NAME libpowerd)
diff --git a/libpowerd/clock.c b/libpowerd/clock.c
index b7df606..395fddc 100644
--- a/libpowerd/clock.c
+++ b/libpowerd/clock.c
@@ -31,7 +31,7 @@
 #include <stdio.h>
 #include <glib.h>
 
-#include "debug.h"
+#include "powerd_debug.h"
 
 #define NSEC_PER_SEC 1000000000L
 #define NSEC_PER_MSEC 1000000L
diff --git a/libpowerd/commands.c b/libpowerd/commands.c
index 088cfc4..11c5097 100644
--- a/libpowerd/commands.c
+++ b/libpowerd/commands.c
@@ -29,7 +29,7 @@
 #include <string.h>
 #include <assert.h>
 #include <glib.h>
-#include <cjson/json.h>
+#include <json.h>
 #include <luna-service2/lunaservice.h>
 
 #include "powerd.h"
diff --git a/libpowerd/init.c b/libpowerd/init.c
index 91f8bff..5a14210 100644
--- a/libpowerd/init.c
+++ b/libpowerd/init.c
@@ -25,7 +25,7 @@
 #include <luna-service2/lunaservice.h>
 
 #include "init.h"
-#include "debug.h"
+#include "powerd_debug.h"
 
 void _PowerdClientIPCRun(void);
 void _PowerdClientIPCStop(void);
diff --git a/libpowerd/wait.c b/libpowerd/wait.c
index a57168e..5f93742 100644
--- a/libpowerd/wait.c
+++ b/libpowerd/wait.c
@@ -33,7 +33,7 @@
 
 #include "wait.h"
 #include "clock.h"
-#include "debug.h"
+#include "powerd_debug.h"
 
 #if !defined(_GNU_SOURCE)
 #endif
diff --git a/powerd/CMakeLists.txt b/powerd/CMakeLists.txt
index 201f8a1..262f4fb 100644
--- a/powerd/CMakeLists.txt
+++ b/powerd/CMakeLists.txt
@@ -28,7 +28,7 @@ file(GLOB SOURCE_FILES alarms/* charging/* suspend/* utils/* *.c)
 webos_configure_source_files(SOURCE_FILES config.c timesaver.c)
 
 add_executable(powerd ${SOURCE_FILES})
-target_link_libraries(powerd ${GLIB2_LDFLAGS} ${LUNASERVICE2_LDFLAGS} ${CJSON_LDFLAGS} ${NYXLIB_LDFLAGS} pthread rt)
+target_link_libraries(powerd ${GLIB2_LDFLAGS} ${LUNASERVICE2_LDFLAGS} ${JSON_LDFLAGS} ${NYXLIB_LDFLAGS} pthread rt)
 
 webos_build_daemon()
 webos_build_system_bus_files()
diff --git a/powerd/alarms/timeout_alarm.c b/powerd/alarms/timeout_alarm.c
index bf97517..4dd3959 100644
--- a/powerd/alarms/timeout_alarm.c
+++ b/powerd/alarms/timeout_alarm.c
@@ -32,7 +32,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdbool.h>
-#include <cjson/json.h>
+#include <json.h>
 #include <sys/stat.h>
 #include <unistd.h>
 #include <luna-service2/lunaservice.h>
diff --git a/powerd/charging/battery.c b/powerd/charging/battery.c
index 579fa17..d607d3c 100644
--- a/powerd/charging/battery.c
+++ b/powerd/charging/battery.c
@@ -28,11 +28,11 @@
 #include <stdbool.h>
 #include <unistd.h>
 #include <glib.h>
-#include <cjson/json.h>
+#include <json.h>
 #include <luna-service2/lunaservice.h>
 
 #include "init.h"
-#include "debug.h"
+#include "powerd_debug.h"
 #include "logging.h"
 #include "main.h"
 #include "battery.h"
diff --git a/powerd/charging/charger.c b/powerd/charging/charger.c
index 9dbd276..6c0ce76 100644
--- a/powerd/charging/charger.c
+++ b/powerd/charging/charger.c
@@ -30,11 +30,11 @@
 #include <stdbool.h>
 #include <unistd.h>
 #include <glib.h>
-#include <cjson/json.h>
+#include <json.h>
 #include <luna-service2/lunaservice.h>
 
 #include "init.h"
-#include "debug.h"
+#include "powerd_debug.h"
 #include "logging.h"
 #include "main.h"
 #include "batterypoll.h"
diff --git a/powerd/charging/suspend.c b/powerd/charging/suspend.c
index f057a29..df53692 100644
--- a/powerd/charging/suspend.c
+++ b/powerd/charging/suspend.c
@@ -32,7 +32,7 @@
 #include <stdlib.h>
 
 #include <syslog.h>
-#include <cjson/json.h>
+#include <json.h>
 
 #include "utils/sysfs.h"
 
@@ -42,7 +42,7 @@
 #include "clock.h"
 #include "config.h"
 #include "wait.h"
-#include "debug.h"
+#include "powerd_debug.h"
 #include "main.h"
 #include "timesaver.h"
 #include "timersource.h"
diff --git a/powerd/main.c b/powerd/main.c
index 949e1dd..76da288 100644
--- a/powerd/main.c
+++ b/powerd/main.c
@@ -29,7 +29,7 @@
 #include <luna-service2/lunaservice.h>
 
 #include "init.h"
-#include "debug.h"
+#include "powerd_debug.h"
 #include "timesaver.h"
 #include "logging.h"
 
diff --git a/powerd/suspend/shutdown.c b/powerd/suspend/shutdown.c
index 02bdff1..73bf475 100644
--- a/powerd/suspend/shutdown.c
+++ b/powerd/suspend/shutdown.c
@@ -34,12 +34,12 @@
 */
 
 #include <glib.h>
-#include <cjson/json.h>
+#include <json.h>
 #include <syslog.h>
 #include <luna-service2/lunaservice.h>
 
 #include "lunaservice_utils.h"
-#include "debug.h"
+#include "powerd_debug.h"
 #include "init.h"
 #include "main.h"
 #include "logging.h"
diff --git a/powerd/suspend/suspend_ipc.c b/powerd/suspend/suspend_ipc.c
index 65c2bc5..8c0366a 100644
--- a/powerd/suspend/suspend_ipc.c
+++ b/powerd/suspend/suspend_ipc.c
@@ -38,11 +38,11 @@
 
 #include <syslog.h>
 #include <glib.h>
-#include <cjson/json.h>
+#include <json.h>
 
 #include "wait.h"
 #include "main.h"
-#include "debug.h"
+#include "powerd_debug.h"
 #include "suspend.h"
 #include "logging.h"
 #include "lunaservice_utils.h"
diff --git a/powerd/utils/init.c b/powerd/utils/init.c
index 06a1396..e536d5e 100644
--- a/powerd/utils/init.c
+++ b/powerd/utils/init.c
@@ -22,7 +22,7 @@
 #include <string.h>
 
 #include "init.h"
-#include "debug.h"
+#include "powerd_debug.h"
 
 typedef struct _GNamedHookList
 {
diff --git a/powerd/utils/sysfs.c b/powerd/utils/sysfs.c
index abf6bff..7d83ade 100644
--- a/powerd/utils/sysfs.c
+++ b/powerd/utils/sysfs.c
@@ -35,7 +35,7 @@
 #include <glib.h>
 
 #include "logging.h"
-#include "debug.h"
+#include "powerd_debug.h"
 
 #define LOG_DOMAIN "sysfs:"
 
diff --git a/powerd/utils/uevent.c b/powerd/utils/uevent.c
index 65d2fb1..9d003f3 100644
--- a/powerd/utils/uevent.c
+++ b/powerd/utils/uevent.c
@@ -33,7 +33,7 @@
 #include <glib.h>
 
 #include "uevent.h"
-#include "debug.h"
+#include "powerd_debug.h"
 
 static gboolean
 ChangeGIOHelper(GIOChannel *source, GIOCondition condition, gpointer ctx)
-- 
2.2.1

